File: 1-Two-Sum/1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> numMap = new Dictionary<int,int>();
        for(int i = 0; i < nums.Length; i++){
            int diff = target - nums[i];
            if(numMap.ContainsKey(diff)){
                return [i, numMap[diff]];
            }
            numMap[nums[i]] = i;
        }
        return [];
    }
}

File: 1-Two-Sum/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            if(numMap.containsKey(diff)){
                return new int[]{i, numMap.get(diff)};
            }
            numMap.put(nums[i], i);
        }
        return new int[]{};
    }
}